cmd/snap-seccomp/main.go

```diff
@@ -0,0 +1,667 @@
+// 32bit userspace).
+func addSecondaryArches(secFilter *seccomp.ScmpFilter) error {
+	// note that all architecture strings are in the dpkg
+	// architecture notation
+	var compatArch seccomp.ScmpArch
+
+	// common case: kernel and userspace have the same arch. We
+	// add a compat architecture for some architectures that
+	// support it, e.g. on amd64 kernel and userland, we add
+	// compat i386 syscalls.
+	if archUbuntuArchitecture() == archUbuntuKernelArchitecture() {
+		switch archUbuntuArchitecture() {
+		case "amd64":
+			compatArch = seccomp.ArchX86
+		case "arm64":
+			compatArch = seccomp.ArchARM
+		case "ppc64":
+			compatArch = seccomp.ArchPPC
+		}
+	} else {
+		// less common case: kernel and userspace have different archs
+		// so add a compat architecture that matches the kernel. E.g.
+		// an amd64 kernel with i386 userland needs the amd64 secondary
+		// arch added to support specialized snaps that might
+		// conditionally call 64bit code when the kernel supports it.
+		// Note that in this case snapd requests i386 (or arch 'all')
+		// snaps. While unusual from a traditional Linux distribution
+		// perspective, certain classes of embedded devices are known
+		// to use this configuration.
+		compatArch = UbuntuArchToScmpArch(archUbuntuKernelArchitecture())
+	}
+
+	if compatArch != seccomp.ArchInvalid {
+		return secFilter.AddArch(compatArch)
+	}
+
+	return nil
+}
+
+func hasLine(content []byte, needle string) bool {
+	for _, line := range bytes.Split(content, []byte("\n")) {
+		if bytes.Equal(bytes.TrimSpace(line), []byte(needle)) {
+			return true
+		}
+	}
+	return false
+}
+
+func compile(content []byte, out string) error {
+	var err error
+	var secFilter *seccomp.ScmpFilter
+
+	// FIXME: right now complain mode is the equivalent to unrestricted.
+	// We'll want to change this once we seccomp logging is in order.
+	if hasLine(content, "@unrestricted") || hasLine(content, "@complain") {
+		secFilter, err = seccomp.NewFilter(seccomp.ActAllow)
+		if err != nil {
+			return fmt.Errorf("cannot create seccomp filter: %s", err)
+		}
+	} else {
+		secFilter, err = seccomp.NewFilter(seccomp.ActKill)
+		if err != nil {
+			return fmt.Errorf("cannot create seccomp filter: %s", err)
+		}
+
+		if err := addSecondaryArches(secFilter); err != nil {
+			return err
+		}
+
+		scanner := bufio.NewScanner(bytes.NewBuffer(content))
+		for scanner.Scan() {
+			if err := parseLine(scanner.Text(), secFilter); err != nil {
+				return fmt.Errorf("cannot parse line: %s", err)
+			}
+		}
+		if scanner.Err(); err != nil {
+			return err
+		}
+	}
+
+	fout, err := os.Create(out)
+	if err != nil {
+		return err
+	}
+	defer fout.Close()
+
+	if osutil.GetenvBool("SNAP_SECCOMP_DEBUG") {
+		secFilter.ExportPFC(os.Stdout)
+	}
+
+	return secFilter.ExportBPF(fout)
+}
+
+func showSeccompLibraryVersion() error {
+	major, minor, micro := seccomp.GetLibraryVersion()
+	fmt.Fprintf(os.Stdout, "seccomp version: %d.%d.%d\n", major, minor, micro)
```

Comment:
I think we can print just the version itself here, since the command is already super specific (library-version).